src/components/dictionaryConcepts/containers/EditConcept.jsx

```diff
@@ -187,6 +199,76 @@ export class EditConcept extends Component {
     }
   }
 
+  addMappingRow = () => {
+    const { mappings } = this.state;
+    mappings.push({
+      map_type: 'Same as',
+      source: null,
+      to_concept_code: null,
+      to_concept_name: null,
+      id: mappings.length + 1,
+      to_source_url: null,
+      isNew: true,
+    });
+    this.setState({ mappings });
+  }
+
+  updateEventListener = (event) => {
+    const { tabIndex, name, value } = event.target;
+    const { mappings } = this.state;
+    mappings[tabIndex][name] = value;
+    if (name !== INTERNAL_MAPPING_DEFAULT_SOURCE && mappings[tabIndex].to_concept_code === null) {
+      mappings[tabIndex].to_concept_code = String(uuid());
+    }
+    this.setState(mappings);
+  }
+
+  updateAsyncSelectValue = (value) => {
+    const { mappings } = this.state;
+    if (value !== null && value.index !== undefined) {
+      mappings[value.index].to_source_url = value.value;
+      mappings[value.index].to_concept_name = value.label;
+    }
+    this.setState({ mappings });
+  }
+
+  removeMappingRow = (event) => {
+    const { tabIndex } = event.target;
+    const { mappings } = this.state;
+    delete mappings[tabIndex];
+    this.setState({ mappings });
+  }
+
+  organizeMappings = (mappings) => {
+    const filteredMappings = mappings.map((mapping, i) => {
+      if (mapping.to_source_url === '/orgs/CIEL/sources/CIEL/') {
+        return {
+          id: i,
+          map_type: mapping.map_type,
+          source: 'CIEL',
```

Comment:
Don't we have a constant for CIEL?

Reply:
@dkayiwa Let me fix that and get back to you.

Reply:
@dkayiwa, I have already made changes.